docs/reference/painless-api-reference.asciidoc

```diff
@@ -0,0 +1,17 @@
+["appendix",role="exclude",id="painless-api-reference"]
+= Painless API Reference
+
+<<modules-scripting-painless,Painless>> has a strict whitelist for methods and
+classes to make sure that all painless scripts are secure and fast. Most of
+these methods are exposed directly from the JRE while others are part of
+Elasticsearch or Painless itself. Below is a list of all available methods
+grouped under the classes on which you can call them. Clicking on the method
+name takes you to the documentation for the method.
+
+NOTE: Methods defined in the JRE also have a `(java 9)` link which can be used
+to see the method's documentation in Java 9 while clicking on the method's name
+goes to the Java 8 documentation. Usually these aren't different but it is
+worth going to the version that matches the version of Java you are using to
+run Elasticsearch just in case.
+
+include::painless-api-reference/index.asciidoc[]
```

Comment:
This file is maintained by hand. Everything in the `painless-api-reference` directory, including `index.asciidoc` is generated.